### PR TITLE
REPO-907: Feature/additional doi autocomplete fields

### PR DESCRIPTION
**Pull request by prdanelli**

### CODE REVIEW

File: app/assets/javascripts/hyku_addons/autofill_work_form.js [hunk 1]
```diff
@@ -76,16 +90,16 @@ class AutofillWorkForm {
       })
 
     } else if ($.type(value) == "object") {
-      var wrapper = this.wrapper(field, index)
+      var $wrapper = this.wrapper(field, index)
 
       // This isn't using the normal setValue method because of the requirement to autofill nested groups of fields
       Object.entries(value).forEach(([childField, childValue]) => {
-        $(wrapper.find($(this.inputSelector(childField))).find(this.targetInputSelector).get(0)).val(childValue)
+        $($wrapper.find($(this.inputSelector(childField))).find(this.targetInputSelector).get(0)).val(childValue)
```
Comment (okeen) [inline app/assets/javascripts/hyku_addons/autofill_work_form.js:97]: This gets confusing to read / understand

File: app/assets/javascripts/hyku_addons/autofill_work_form.js [hunk 2]
```diff
@@ -94,6 +108,44 @@ class AutofillWorkForm {
...
+
+    funders.forEach((funder, i) => {
+      // Set the parent fields container for each iteration
+      var $wrapper = this.wrapper("funder", i)
+
+      Object.entries(funder).forEach(function([childField, childValue]){
+        if ($.type(childValue) === "array") {
+          childValue.forEach(function(val, j) {
+            // We need to get this fresh each time as we are adding DOM elements
+            $($wrapper.find($(this.inputSelector(childField))).find(this.targetInputSelector).get(j)).val(val)
```
Comment (okeen) [inline app/assets/javascripts/hyku_addons/autofill_work_form.js:127]: I would split this into simpler sentences for the sake of maintanability

File: app/assets/javascripts/hyku_addons/autofill_work_form.js (hunk 1 above)
Comment (prdanelli) [reply app/assets/javascripts/hyku_addons/autofill_work_form.js:97]: Yea, its a little scruffy. I'll tidy that up now.

File: app/services/bolognese/writers/hyku_addons_work_form_fields_writer.rb [hunk 3]
```diff
@@ -4,37 +4,38 @@
...
-          "language" => Array(language)
-        }.compact.reject { |_key, value| value.blank? }
+      def hyku_addons_work_form_fields(curation_concern: "generic_work")
+        @curation_concern = curation_concern
+
+        # Work through each of the work types fields to create the data hash
+        form_data = work_type_terms.each_with_object({}) do |term, data|
+          method_name = "write_#{term}"
+
+          data[term.to_s] = respond_to?(method_name, true) ? send(method_name) : nil
```
Comment (okeen) [inline app/services/bolognese/writers/hyku_addons_work_form_fields_writer.rb:31]: I think you can substitue the full sentence for a single `try`:

data[term.to_s] = try(method_name)

**Review by okeen — DISMISSED** [review]

File: app/services/bolognese/writers/hyku_addons_work_form_fields_writer.rb (hunk 3 above)
Comment (prdanelli) [reply app/services/bolognese/writers/hyku_addons_work_form_fields_writer.rb:31]: It seems that `try` uses `public_send` which wont work for protected methods: 

https://github.com/rails/rails/blob/main/activesupport/lib/active_support/core_ext/object/try.rb#L7

**Review by okeen — APPROVED** [review]

**Pull request merged**